src/modules/tests/tests.reducer.ts

```diff
@@ -89,55 +63,18 @@ const deriveSlotId = (state: TestsModel, action: Action): string | null => {
   return {
     ...state,
     startedTests: {
       ...state.startedTests,
       [slotId]: {
         ...state.startedTests[slotId],
-        // Each sub-reducer deals with state scoped to a specific test and has no knowledge of
-        // the context of which test contains it that state.
-        // Here, combineReducers delegates management of the sub-state navigated here for a given
-        // slotId to the relevant sub-reducer
-        ...nestedCombineReducers(
-          {
-            version: schemaVersionReducer,
-            category: categoryReducer,
-            journalData: {
-              examiner: examinerReducer,
-              testCentre: testCentreReducer,
-              testSlotAttributes: testSlotsAttributesReducer,
-              candidate: candidateReducer,
-              applicationReference: applicationReferenceReducer,
-            },
-            preTestDeclarations: preTestDeclarationsReducer,
-            accompaniment: accompanimentReducer,
-            vehicleDetails: vehicleDetailsReducer,
-            instructorDetails: instructorDetailsReducerFactory(get(state.startedTests[slotId], 'category')),
-            testData: testDataReducerFactory(get(state.startedTests[slotId], 'category')),
-            passCompletion: passCompletionReducer,
-            postTestDeclarations: postTestDeclarationsReducer,
-            testSummary: testSummaryReducer,
-            communicationPreferences: communicationPreferencesReducer,
-            rekey: rekeyReducer,
-            rekeyDate: rekeyDateReducer,
-            rekeyReason: rekeyReasonReducer,
-            examinerBooked: examinerBookedReducer,
-            examinerConducted: examinerConductedReducer,
-            examinerKeyed: examinerKeyedReducer,
-            changeMarker: changeMarkerReducer,
-          }, combineReducers,
-        )(
-          // The redux pattern necessitates that the state tree be initialised
-          // with all its properties declared. This conflicts with the
-          // 'StandardCarTestCATBSchema' TS interface as many of its properties are optional (?).
-          // In order to reconcile the TS interface and the redux reducer pattern we use
-          // the TS 'Required' mapped type which The 'Required' type which strips ? modifiers
-          // from all properties of 'StandardCarTestCATBSchema', thus making all properties required.
-          state.startedTests[slotId] as Required<StandardCarTestCATBSchema>,
-          action,
-        ),
+        // TODO - Need a way to get category - it needs to be in all the start test actions
+        // or we need to select it from the journal state
+        // ...testsReducerFactory('B', action, state.startedTests[slotId]),
+        // ...testsReducerFactory('B+E', action, state.startedTests[slotId]),
+        ...testsReducerFactory(get(state.startedTests[slotId], 'category'), action, state.startedTests[slotId]),
```

Comment:
Category should be simple to plumb through no?
And blimey, this is cool! Have we tested that this won't balloon our reducer calls when actions are dispatched?

Reply:
It will be, just need to make a decision on how to plumb it through, I am not really keen and putting a selector into the journal module here. So I think we need to do one of the following:

1) Refactor all the start test actions to send a category
2) Move the current start test actions to hang of the set category action so start test -> set category -> all other actions

Reply:
First one makes sense, would the second one involve effects? I'd really like to avoid them!

Reply:
The whole Start Test process is based on effects , so you can't avoid them

Reply:
but yea the second idea involves changing the effects